### ISSUE TITLE
Swagger plugin incorrectly parses optional fields

### ISSUE THREAD

**Issue by svmn** [nestjs/swagger#563]
## I'm submitting a...

<pre><code>
[x] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Minimal reproduction of the problem with instructions

1. Enable "@nestjs/swager/plugin"
2. Create following DTO
```
OptonalPropDto {
  field?: string;
}
```
3. Check output swagger document
```
            "OptonalPropDto ": {
                "type": "object",
                "properties": {
                    "field": {
                        "type": "object"
                    }
                }
            }
```
Field type should have "string" value.

## Environment
Downgrading @nestjs/cli helps to resolve the issue.
<pre><code>
"@nestjs/swagger": "4.2.8"
"@nestjs/cli": "6.14.2"

 
For Tooling issues:
- Node version: 12.15
- Platform:  Windows

Others:

</code></pre>

**Comment by ShayDavidson** [comment]
@svmn to which `@nestjs/cli` version did you have to downgrade? facing the same issue

**Comment by kamilmysliwiec** [comment]
Please, provide a minimal repository which reproduces your issue.

**Comment by svmn** [comment]
@kamilmysliwiec Please check https://github.com/svmn/nestjs-swagger-issue

@ShayDavidson Initially it was working with @nestjs/cli:6.10.5, but downgrading not always helps for some reason.

**Comment by kamilmysliwiec** [comment]
Fixed in 4.2.9. This was caused by the `strict` option in the `tsconfig` file

**Comment by svmn** [comment]
Unfortunately, doesn't work with an optional enum
```
enum OptionalEnum {
  ONE,
  TWO,
  THREE,
}

export class HelloResponseDto {
  public optionalProperty?: OptionalEnum;
}
```
still reports type as "object".
using @nestjs/swagger:4.2.9

**Comment by kamilmysliwiec** [comment]
@svmn Can you test with 4.2.10?

**Comment by svmn** [comment]
Works now, thanks.

**Comment by svmn** [comment]
Sorry for being annoying, but another issue occurred now. 
Consider the following DTOs:

```
export class OptionalNestedDto {
  prop: string;
}

export class HelloResponseDto {
  public optionalProperty?: OptionalNestedDto;
}
```

reports error on build:
```
(node:20200) UnhandledPromiseRejectionWarning: ReferenceError: any is not defined
    at Function._OPENAPI_METADATA_FACTORY (C:\Users\user\Documents\nestjs-swagger\dist\hello.dto.js:12:115)
    at ModelPropertiesAccessor.applyMetadataFactory (C:\Users\user\Documents\nestjs-swagger\node_modules\@nestjs\swagger\dist\services\model-properties-accessor.js:26:93)
    at SchemaObjectFactory.exploreModelSchema (C:\Users\user\Documents\nestjs-swagger\node_modules\@nestjs\swagger\dist\services\schema-object-factory.js:54:38)
    at ResponseObjectFactory.create (C:\Users\user\Documents\nestjs-swagger\node_modules\@nestjs\swagger\dist\services\response-object-factory.js:45:47)    
    at lodash_1.mapValues (C:\Users\user\Documents\nestjs-swagger\node_modules\@nestjs\swagger\dist\explorers\api-response.explorer.js:50:96)
    at C:\Users\user\Documents\nestjs-swagger\node_modules\lodash\lodash.js:13401:38
    at C:\Users\user\Documents\nestjs-swagger\node_modules\lodash\lodash.js:4905:15
    at baseForOwn (C:\Users\user\Documents\nestjs-swagger\node_modules\lodash\lodash.js:2990:24)
    at Function.mapValues (C:\Users\user\Documents\nestjs-swagger\node_modules\lodash\lodash.js:13400:7)
    at mapResponsesToSwaggerResponses (C:\Users\user\Documents\nestjs-swagger\node_modules\@nestjs\swagger\dist\explorers\api-response.explorer.js:50:39)   
(node:20200) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:20200) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Compiled code:
```
    static _OPENAPI_METADATA_FACTORY() {
        return { optionalProperty: { required: false, type: () => require("./hello.dto").OptionalNestedDto, enum: any } };
    }
```

For some reason, referenced DTO treated as an enum.

**Comment by kamilmysliwiec** [comment]
It should be fixed in 4.2.11
